styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java

```diff
@@ -153,11 +139,7 @@ void writeEvent(SequenceEvent sequenceEvent) throws IOException {
       throws IOException {
     final Set<SequenceEvent> storedEvents = readEvents(workflowInstance);
     final Optional<SequenceEvent> lastStoredEvent = storedEvents.stream().reduce((a, b) -> b);
```

Comment:
This way of getting the last element feels quite brutal. ;)

Maybe something like `return Optional.ofNullable(Iterators.getLast(storedEvents.iterator(), null)).map(SequenceEvent::counter);`?